plugins/modules/net_tools/dnsimple.py

```diff
@@ -150,9 +148,10 @@
 
 DNSIMPLE_IMP_ERR = None
 try:
-    from dnsimple import DNSimple
-    from dnsimple.dnsimple import __version__ as dnsimple_version
-    from dnsimple.dnsimple import DNSimpleException
+    from dnsimple import Client, DNSimpleException
+    from dnsimple.service import Accounts
+    from dnsimple.version import version as dnsimple_version
+    from dnsimple.struct.zone_record import ZoneRecordUpdateInput, ZoneRecordInput
```

Comment:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, add tests for this change.

Reply:
I created the changelog fragment. However, I am not sure how and where I should implement the tests, Would this be a unit test as described [here](https://docs.ansible.com/ansible/latest/dev_guide/testing_units_modules.html)?

Reply:
Right now the module has no tests at all, so you would have to create tests from scratch. If you want to implement tests for this module, this definitely would be great (also since it prevents some future breakages to go undetected until someone tries to use the module :) ). The best would be to add unit tests.

I'm not sure whether the link you have actually considers collections at all. Here are some very basic for unit testing in collections: https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_testing.html#adding-unit-tests The tests would go in this directory as `test_dnsimple.py`: https://github.com/ansible-collections/community.general/tree/main/tests/unit/plugins/modules/net_tools

A big question is whether you want to mock the API itself, or mock the dnsimple Python library. Both have upsides and downsides...